contracts/reserve/GoodReserveCDai.sol

```diff
@@ -275,25 +128,26 @@ contract GoodReserveCDai is
 	 * @param _targetAddress address of g$ and gdx recipient if different than msg.sender
 	 * @return (gdReturn) How much GD tokens were transferred
 	 */
-	function _buy(
+	function buy(
 		uint256 _tokenAmount,
 		uint256 _minReturn,
 		address _targetAddress
-	) internal returns (uint256) {
+	) external returns (uint256) {
 		ERC20 buyWith = ERC20(cDaiAddress);
```

Comment:
if targetAddress is zero it should be set to msg.sender

Reply:
Changed